src/sentry/api/endpoints/project_transaction_threshold.py

```diff
@@ -89,9 +89,10 @@ def post(self, request, project):
                 project_threshold.edited_by = request.user
                 project_threshold.save()
 
-                created = False
+            created = False
 
-            except ProjectTransactionThreshold.DoesNotExist:
+        except ProjectTransactionThreshold.DoesNotExist:
+            with transaction.atomic():
```

Comment:
Is it possible to hit an integrity error here if the threshold gets created between the `get` from above and the `create` from below?

Reply:
Moved here https://github.com/getsentry/sentry/pull/26965 so this PR isn't cluttered